server/src/main/java/org/candlepin/pinsetter/tasks/ImportJob.java

```diff
@@ -0,0 +1,135 @@
+
+/**
+ * Runs an asynchronous manifest import. This job expects that the manifest file
+ * was already uploaded.
+ */
+public class ImportJob extends UniqueByEntityJob {
+
+    protected static final String STORED_FILE_ID = "stored_manifest_file_id";
+    protected static final String CONFLICT_OVERRIDES = "conflict_overrides";
+    protected static final String UPLOADED_FILE_NAME = "uploaded_file_name";
+
+    private static Logger log = LoggerFactory.getLogger(ImportJob.class);
+
+    private OwnerCurator ownerCurator;
+    private ManifestManager manifestManager;
+
+    @Inject
+    public ImportJob(OwnerCurator ownerCurator, ManifestManager manifestManager) {
+        this.ownerCurator = ownerCurator;
+        this.manifestManager = manifestManager;
+    }
+
+    @Override
+    public void toExecute(JobExecutionContext context) throws JobExecutionException {
+        JobDataMap map = context.getMergedJobDataMap();
+        String ownerKey = (String) map.get(JobStatus.TARGET_ID);
+        ConflictOverrides overrides = new ConflictOverrides((String[]) map.get(CONFLICT_OVERRIDES));
+        String storedFileId = (String) map.get(STORED_FILE_ID);
+        String uploadedFileName = (String) map.get(UPLOADED_FILE_NAME);
+
+        Throwable caught = null;
+        Owner targetOwner = null;
+        try {
+            targetOwner = ownerCurator.lookupByKey(ownerKey);
+            if (targetOwner == null) {
+                //FIXME This should be internationalized.
+                context.setResult("Nothing to do. Owner no longer exists.");
+                return;
+            }
+
+            ImportRecord importRecord = manifestManager.importStoredManifest(targetOwner,
+                storedFileId, overrides, uploadedFileName);
+            context.setResult(importRecord);
+        }
+
+        // Catch and handle SyncDataFormatException and ImporterExceptions
+        // as the OwnerResource.importManifest does which will provide a little more
+        // info about the exception that was thrown (CandlepinException).
+        catch (SyncDataFormatException e) {
+            caught = new BadRequestException(e.getMessage(), e);
+        }
+        catch (ImporterException e) {
+            caught = new IseException(e.getMessage(), e);
+        }
+        catch (Exception e) {
+            caught = e;
+        }
+
+        if (caught != null) {
+            log.error("ImportJob encountered a problem.", caught);
+            manifestManager.recordImportFailure(targetOwner, caught,
+                uploadedFileName);
+            context.setResult(caught.getMessage());
+            // If an exception was thrown, the importer's transaction was rolled
+            // back. We want to make sure that the file gets deleted so that it
+            // doesn't take up disk space. It may be possible that the file was
+            // already deleted, but we attempt it anyway.
+            manifestManager.deleteStoredManifest(storedFileId);
```

Comment:
> looks like importStoredManifest is called only via the importJob , and we delete manifest upon both success and failure. perhaps we could do that in a finally block?
> 
> On the other hand, I can see it being useful for debugging if, upon failure, we could avoid deleting a manifest if explicitly requested as such. thoughts?

The issue here is that we don't want stored manifest files lingering around in the database as we do not want the DB to swell. Even thought they will be cleaned out every 24h (or other configured value), users tend to continuously retry. It is for import though, so I'm not sure how frequent it would be.